ring/ring.go

```diff
@@ -318,10 +314,6 @@ func (r *Ring) loop(ctx context.Context) error {
 }
 
 func (r *Ring) stopping(_ error) error {
```

Comment:
You can completely remove this function and pass `nil` instead of `r.stopping` to `services.NewBasicService()`.

Reply:
Oh yeah, true!